storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -810,9 +810,15 @@ private static int numUsedWorkers(SchedulerAssignment assignment) {
         return ret;
     }
 
+    /**
+     * Check new assignments with existing assignments and determine difference is any.
+     *
+     * @param existingAssignments non-null map of topology-id to existing assignments.
+     * @param newAssignments non-null map of topology-id to new assignments.
+     * @return true if there is a change in assignments, false otherwise.
+     */
     private boolean auditAssignmentChanges(Map<String, Assignment> existingAssignments,
                                            Map<String, Assignment> newAssignments) {
-        assert existingAssignments != null && newAssignments != null;
```

Comment:
why remove these checks?

Reply:
The two arguments are never null - and this is a private method - with javadoc that says it should be non-null. See calling method.

If null is passed in some future code, then NullPointerException will be thrown automatically. I can't think of alternative replacement code that will behave differently, unless this method allows null, and null is treated as an empty assignment.

Reply:
sounds good